messaging/src/test/java/org/axonframework/eventhandling/ReplayTokenTest.java

```diff
@@ -135,4 +154,1470 @@ void createReplayTokenReturnsReplayToken() {
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 9 was processed before reset, should be marked as replay even though gaps were filled");
+        }
+
+        @Test
+        void eventAtResetIndexShouldBeReplayEvenWhenGapsFilledDuringReplay() {
+            // tokenAtReset at index 10 with gaps at 7,8
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, null);
+
+            // During replay, gaps get filled, process 0-9
+            // Event 10 - at reset index, WAS processed before reset
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(10, emptySet());
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode at reset index");
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 10 was processed before reset, should be marked as replay");
+        }
+
+        @Test
+        void replayBeforeResetIndexAdnGaps() {
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, null);
+
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(6, emptySet());
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertTrue(ReplayToken.isReplay(result));
+        }
+
+        @Test
+        void replayAtGapAndWithAnotherGap() {
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, null);
+
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(8, setOf(7L));
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertFalse(ReplayToken.isReplay(result));
+        }
+
+        @Test
+        void replayBeforeResetIndexEvenWhenGapsWereFilledDuringReplay() {
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, null);
+            for (int i = 0; i <= 9; i++) {
+                currentToken = ((ReplayToken) currentToken).advancedTo(
+                        GapAwareTrackingToken.newInstance(i, emptySet())
+                );
+            }
+
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(10, emptySet());
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertTrue(ReplayToken.isReplay(result));
+        }
+
+        @Test
+        void replayStartsBeforeResetIndexWithoutGaps() {
```

Comment:
Isn't this test method identical to `eventPreviouslyProcessedShouldBeReplayEvenWhenGapsFilledDuringReplay`?